src/test/ui/rfc-2632-const-trait-impl/tilde-const-and-const-params.rs

```diff
@@ -0,0 +1,34 @@
+// check-pass
+#![feature(const_trait_impl)]
+#![feature(generic_arg_infer)]
+#![feature(generic_const_exprs)]
+#![allow(incomplete_features)]
+
+struct Foo<const N: usize>;
+
+impl<const N: usize> Foo<N> {
+   fn add<A: ~const Add42>(self) -> Foo<{ A::add(N) }> {
```

Comment:
This can't be sound. `~const` explicitly mean `const` but *ehh*. So it could be not const. We need something like `A: const Trait` which is "always const" to make this work. I will open a revert PR shortly.

Reply:
I just came here because I broke this test in https://github.com/rust-lang/rust/pull/101900 😆

so yea, this is unsound or depending on the impl, very confusing, as making a function `const fn` will make it stop compiling